src/internal/SignalGenerator.cpp

```diff
@@ -0,0 +1,47 @@
+/**
+ * @file SignalGenerator.cpp
+ * @author Luciana Moreno Borges (lumoreno.borges@gmail.com)
+ * @brief Signal Generator class
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <bits/stdc++.h>
+#include <chrono>
+#include <../src/internal/SignalGenerator.h>
+
+namespace hyro
+{
+  /**
+   * @brief Construct a new Signal Generator:: Signal Generator object
+   * 
+   */
+SignalGenerator::SignalGenerator():
+  m_deg(0)
+{
+}
+
+ 
+ /* Get the sine or cos for the desired angle */
+float SignalGenerator::getSignalValue(Function func)
+{
+  float rad = m_deg / 180.0 * M_PI;
+  m_deg += m_step;
+
+  return (cosine) ? m_amp*cos(rad) : m_amp*sin(rad);
```

Comment:
What do you do with the `func` argument in this method?
Don't you need to check it?